packages/data-broker/src/services/dhis/pullers/DeprecatedEventsPuller.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { EventsPuller } from './EventsPuller';
+
+/**
+ * This is a deprecated puller which invokes a slow DHIS2 api ('/events')
+ * and returns an obsolete data structure (equivalent to the raw DHIS2 events).
+ * It is invoked using the `options.useDeprecatedApi` flag
+ *
+ * TODO Delete this puller as soon as all its past consumers have migrated over to
+ * the new (non-deprecated) method
+ */
+export class DeprecatedEventsPuller extends EventsPuller {
```

Comment:
Split this out into its own class because EventsPuller.test.js was too long and hard to read.